crates/biome_js_analyze/src/lint/nursery/use_consistent_object_definition.rs

```diff
@@ -0,0 +1,238 @@
+    /// Using the same style consistently across your codebase makes it easier to quickly read and understand object definitions.
+    ///
+    /// ## Example
+    ///
+    /// ### Invalid
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "syntax": "explicit"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// let foo = 1;
+    /// let invalid = {
+    ///     foo
+    /// };
+    /// ```
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// let invalid = {
+    ///     bar() { return "bar"; },
+    /// };
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js,use_options
+    /// let foo = 1;
+    /// let valid = {
+    ///     foo: foo,
+    ///     bar: function() { return "bar"; },
+    /// };
+    /// ```
+    ///
+    /// ### Invalid
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "syntax": "shorthand"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// let foo = 1;
+    /// let invalid = {
+    ///     foo: foo
+    /// };
+    /// ```
+    ///
+    /// ```js,expect_diagnostic,use_options
+    /// let invalid = {
+    ///     bar: function() { return "bar"; },
+    /// };
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js,use_options
+    /// let foo = 1;
+    /// let valid = {
+    ///     foo,
+    ///     bar() { return "bar"; },
+    /// };
+    /// ```
+    ///
+    /// ## Options
+    ///
+    /// Use the options to specify the syntax of object literals to enforce.
+    ///
+    /// ```json,options
+    /// {
+    ///     "options": {
+    ///         "syntax": "explicit"
+    ///     }
+    /// }
+    /// ```
+    ///
+    /// ### syntax
+    ///
+    /// The syntax to use:
+    /// - `explicit`: enforces the use of explicit object property syntax in every case
+    /// - `shorthand`: enforces the use of shorthand object property syntax when possible
+    ///
+    /// **Default:** `explicit`
+    ///
+    pub UseConsistentObjectDefinition {
+        version: "next",
+        name: "useConsistentObjectDefinition",
+        language: "js",
+        recommended: false,
+        severity: Severity::Error,
+        sources: &[RuleSource::Eslint("object-shorthand")],
+        source_kind: RuleSourceKind::Inspired,
+    }
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields, default)]
+pub struct UseConsistentObjectDefinitionOptions {
+    syntax: ObjectPropertySyntax,
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Deserializable, Eq, PartialEq, Serialize)]
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub enum ObjectPropertySyntax {
+    /// {foo: foo}
```

Comment:
May be good to put backticks around this:

```suggestion
    /// `{ foo: foo }`
```

(Same for the other one.)

Reply:
Done! I also now instantly return a result instead of storing the name_token, in the 2nd top `match`. The linter was (rightfully) complaining that I was storing it but never actually using it. It was a leftover from a refactor of my previous implementation.